Add http-1x examples for server SDK usage

drganjoo opened the pull request:
This commit updates examples to demonstrate server SDK usage with both http@0.2.x
and http@1.x, supporting the http-1x migration effort.

## Changes

### New Directories
- **legacy/**: Complete set of examples using http@0.2.x/hyper@0.14 with
  aws-smithy-legacy-http-server. These examples preserve the original behavior
  for users not yet ready to migrate.

- **pokemon-service-server-skd-patch/**: Compatibility layer and patches for
  bridging http@0 and http@1 types in server SDK.

### Updated Examples
Modified existing examples to use http@1.x/hyper@1.x:
- pokemon-service: Main service example updated for http@1
- pokemon-service-tls: TLS example with hyper 1.x compatibility
- pokemon-service-lambda: Lambda example updated for http@1
- pokemon-service-client-usage: Client examples updated
- pokemon-service-common: Common utilities updated

### Key Updates
- Cargo.toml files updated with http@1.x and hyper@1.x dependencies
- Server initialization code updated for hyper 1.x API changes
- Test helpers updated to support both HTTP versions
- Connector and middleware examples updated for new hyper API

## Testing
Examples can be run with both HTTP versions to verify compatibility.

Relates to #3362

ysaito1001 committed:
Revert version 2.1 of the IMDS credentials provider (#4187)

## Motivation and Context
We released the feature [in this
release](https://github.com/awslabs/aws-sdk-rust/releases/tag/release-2025-06-12),
but till the infrastructure is fully ready, we'll temporarily revert the
functionality.

## Testing
- CI

## Checklist
- [x] For changes to the AWS SDK, generated SDK code, or SDK runtime
crates, I have created a changelog entry Markdown file in the
`.changelog` directory, specifying "aws-sdk-rust" in the `applies_to`
key.

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

ysaito1001 committed:
Make Rpc V2 CBOR `awsQuery` compatible (#4186)

## Description
This PR makes Rpc V2 CBOR a compatible protocol for `awsQuery` using
`awsQueryCompatible` trait, as described in the `Important` section in
[this
page](https://smithy.io/2.0/aws/protocols/aws-query-protocol.html#aws-protocols-awsquerycompatible-trait).

Previously, the implementation for `awsQueryCompatible` was tightly
coupled to `awsJson1_0`. This PR makes the implementation a bit more
abstract so that the implementation can support more target protocols
generically.

## Testing
- CI
- Made `AwsQueryCompatibleTest` parameterized tests to support RpcV2Cbor
and verified against modified cloudwatch service model (with
`awsQueryCompatible` trait applied). ~Tests for RpcV2Cbor is commented
out till https://github.com/smithy-lang/smithy-rs/pull/4185 is merged~.

## Checklist
<!--- If a checkbox below is not applicable, then please DELETE it
rather than leaving it unchecked -->
- [x] For changes to the smithy-rs codegen or runtime crates, I have
created a changelog entry Markdown file in the `.changelog` directory,
specifying "client," "server," or both in the `applies_to` key.
- [x] For changes to the AWS SDK, generated SDK code, or SDK runtime
crates, I have created a changelog entry Markdown file in the
`.changelog` directory, specifying "aws-sdk-rust" in the `applies_to`
key.

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Http-1x update phase 1 (#4181)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
The first steps to defaulting to `http-1x` dependencies instead of
`http-02x`

## Description
<!--- Describe your changes in detail -->
For this PR I updated the dependencies on SDK crates to enable their
`http-1x` features and worked backwards from there fixing the things
that broke (this would be easier to see if the diff was working but it
seems to be broken for feature branches after the new ECR update). This
involved:
* Updating the `http_request_checksum` inlineable to use `http-1x` types
(and removing the `http-1x` feature since it is no longer needed and
wasn't a "real" feature to begin with
* Creating a new `pub mod content_encoding_http_1x` module in the
`aws-runtime` crate. Most logic here is dedicated to implementing the
`http_body_1x::Body` trait for `AwsChunkedBody`. I initially considered
keeping this in the existing `content_encoding` module, but with the
current feature flags and all of the changes I had to make it felt
cleaner to break it out.
* Update body polling logic for `ChecksumBody` calculation and
validation for `http-1x` types


## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran
to -->
<!--- see how your change affects other areas of the code, etc. -->
Updated existing tests and added new ones for `http-1x` types

## TODOs
Future PRs will
* Update code generation logic in codegen-core to use `http-1x`
* Remove pre-1.0 `http` dependencies (anywhere they aren't re-exported)
* Normalize dependency names to `http-1x`, `http-02x`, etc

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Update SdkBody to support http-1x bodies (#4199)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
Continuing the work to default to `http-1x` types under the hood. 

## Description
<!--- Describe your changes in detail -->
This is unfortunately an enormous PR (at least in terms of files
touched, not total lines). The vast majority of changes are just
replacing references to `http_02x` with `http_1x` or `http_body_04x`
with `http_body_1x` and updating code or feature flags to match.

The primary functional changes are in the vicintiy of
`rust-runtime/aws-smithy-types/src/body.rs`, updating methods for
`SdkBody` to understand `http_1x` types. This required several changes:
* Add a new `HttpBody1` variant to the `enum BoxBody`
* Update `SdkBody::from_body_1_x` to construct a `HttpBody1` variant
instead of the `HttpBody04` variant it was previously
* Add a new `trailers: Option<VecDeque<http_1x::HeaderMap>>` field to
`SdkBody` to cache polled trailers (`http_body_1x` doesn't allow polling
trailers separately, but `SdkBody` does, so we have to cache at least
the first trailer frame if it is encountered while polling the body)
* Update `SdkBody::{poll_next, poll_next_trailers}` to work with the new
`HttpBody1x` variant.

## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran
to -->
<!--- see how your change affects other areas of the code, etc. -->
Updated most existing tests to use `http-1x` types (Although I have
almost certainly missed some)

Note that some tests are failing here and that is expected:
* Semver tests: we have changed/deleted some `http_02x` trait impls in
pre-1.0 crates, so we expect these to fail
* Server tests: I haven't touched these and will work with Fahad to get
them updated

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Merge branch 'main' into feature/http-1.x

Note: as of this merge aws-smithy-checksums/src/body/calculate.rs is broken

I need to reconcile the changes from #4181 and #4200, but I am going to
do that in a separate commit so I can get it PRed without all the extraneous
merge from main bits

landonxjames committed:
Aligning http-1x checksum updates with checksum changes from #4200 (#4204)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
There were some conflicts around checksum changes made in
https://github.com/smithy-lang/smithy-rs/pull/4200 while merging main
back into the `feature/http-1.x` branch. This PR reconciles those
changes

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Bump crate versions

cargoUpdateAllLockfiles

Fix sigv4 test issues

landonxjames committed:
Bump http dep version in aws-smithy-http-client

landonxjames committed:
Attempt to bump fuzzgen to http-1x

landonxjames committed:
Bump http-body-util

landonxjames committed:
Merge current `main` into the `feature/http-1.x` branch (#4281)

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Bump aws-smithy-http to 0.63 per request from server team (#4334)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
Request from @drganjoo 

>Also, in feature/http-1.x could we please bump version of
aws-smithy-http to 0.63.x instead of
[0.62.4](https://github.com/smithy-lang/smithy-rs/blob/feature/http-1.x/rust-runtime/aws-smithy-http/Cargo.toml#L3).
This way on the server side we can keep on relying on 0.62.x to have
http/0.x support for a while. Thanks.

Note: I merged main back into the base `feature/http-1.x` branch as part
of making this change

## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran
to -->
<!--- see how your change affects other areas of the code, etc. -->

Failing tests are all server related and expected since the server side
of the codebase hasn't done this upgrade yet.

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Merge branch 'main' into feature/http-1.x

landonxjames committed:
Bump aws-smithy-json to 0.62 per request from Fahad

landonxjames committed:
Remove features = ["http-body-1-x"] from aws-smithy-types for aws-smithy-json and aws-smithy-cbor

landonxjames committed:
Merge branch 'main' into feature/http-1.x

rcoh commented on examples/upgrade-http1x.md:
```diff
@@ -0,0 +1,645 @@
+
+## Why Upgrade?
+
+- **Better Performance**: Hyper 1.x has improved performance and reduced allocations
```
is this actually true?

drganjoo replied:
Ops! I had completely forgotten to review the file before pushing the commit - sorry.

rcoh commented on examples/upgrade-http1x.md:
````diff
@@ -0,0 +1,645 @@
+
+**After:**
+```rust
+use http::StatusCode;
````
can we re-export this from the server or `aws-smithy-http-server`? I don't see any compelling reason to force a direct `http` dep.

drganjoo replied:
It is being re-exported from the aws-smithy-http-server: rust-runtime/aws-smithy-http-server/src/lib.rs.

Need to change the guide.

rcoh commented on examples/upgrade-http1x.md:
```diff
@@ -0,0 +1,645 @@
+
+## Client-Side Changes
+
+### HTTP Client Connector Setup
```
need to call out that client/server http versions are totally decoupled.

drganjoo replied:
I need to check with the SDK team on this section. For the time being I am removing it.

rcoh commented on examples/upgrade-http1x.md:
````diff
@@ -0,0 +1,645 @@
+
+let status = StatusCode::OK;
+let status = StatusCode::NOT_FOUND;
+```
````
these are already covered elsewhere

ysaito1001 committed:
Merge main into feature/http-1.x (#4391)

## Motivation and Context
Merge the latest main to the `feature/http-1.x` branch, plus replace
http-0.x constructs with http-1.x constructs as described in [Tips for
merging to
feature/http-1.x](https://github.com/smithy-lang/smithy-rs/pull/4384)

## Testing
- CI: ignore server related failures and semver hazard failure (the type
being complained about `Uri` isn't exposed since its enclosing module
`request` isn't exposed as pub, which confuses the semver hazard check)

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

---------

Co-authored-by: Landon James <lnj@amazon.com>
Co-authored-by: Russell Cohen <rcoh@amazon.com>
Co-authored-by: AWS SDK Rust Bot <aws-sdk-rust-primary@amazon.com>
Co-authored-by: AWS SDK Rust Bot <97246200+aws-sdk-rust-ci@users.noreply.github.com>
Co-authored-by: vcjana <vcjana@amazon.com>
Co-authored-by: Jason Gin <67525213+jasgin@users.noreply.github.com>
Co-authored-by: Aaron Todd <aajtodd@users.noreply.github.com>

ysaito1001 committed:
Merge branch main into feature/http-1.x (#4402)

## Motivation and Context
Merge the main branch to the `feature/http-1.x` branch. It takes all
from the main branch except for
[this](https://github.com/smithy-lang/smithy-rs/pull/4392#discussion_r2524200307),
which is replaced by `http-1.x` construct.


----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

landonxjames committed:
Merge main to feature/http-1.x (#4422)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## Description
<!--- Describe your changes in detail -->

## Testing
<!--- Please describe in detail how you tested your changes -->
<!--- Include details of your testing environment, and the tests you ran
to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist
<!--- If a checkbox below is not applicable, then please DELETE it
rather than leaving it unchecked -->
- [ ] For changes to the smithy-rs codegen or runtime crates, I have
created a changelog entry Markdown file in the `.changelog` directory,
specifying "client," "server," or both in the `applies_to` key.
- [ ] For changes to the AWS SDK, generated SDK code, or SDK runtime
crates, I have created a changelog entry Markdown file in the
`.changelog` directory, specifying "aws-sdk-rust" in the `applies_to`
key.

----

_By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice._

---------

Co-authored-by: ysaito1001 <awsaito@amazon.com>
Co-authored-by: Russell Cohen <rcoh@amazon.com>
Co-authored-by: AWS SDK Rust Bot <aws-sdk-rust-primary@amazon.com>
Co-authored-by: AWS SDK Rust Bot <97246200+aws-sdk-rust-ci@users.noreply.github.com>
Co-authored-by: vcjana <vcjana@amazon.com>
Co-authored-by: Jason Gin <67525213+jasgin@users.noreply.github.com>
Co-authored-by: Aaron Todd <aajtodd@users.noreply.github.com>
Co-authored-by: greenwoodcm <greenwd@amazon.com>
Co-authored-by: Jason Gin <jasongin88@gmail.com>
Co-authored-by: Aaron J Todd <todaaron@amazon.com>
Co-authored-by: Anna H <annahay@amazon.com>
Co-authored-by: Ariel Ben-Yehuda <ariel.byd@gmail.com>
Co-authored-by: Ariel Ben-Yehuda <arielby@amazon.com>

drganjoo committed:
aws-smithy-http-server - http/1 support (#4373)

Support for `http@1.x`/`hyper@1.x` in `aws-smithy-http-server` runtime crate for the server.

## Key Changes
### 1. Dependency Upgrades
- **http**: `0.2.9` → `1.x`
- **http-body**: `0.4.5` → `1.0`
- **hyper**: `0.14.26` → `1.x` (with `server`, `http1`, `http2`
features)
- **hyper-util**: Added `0.1` for server utilities (`server-auto`, `server-graceful`, etc.)
- **http-body-util**: Added `0.1` for body utilities
- **tower-http**: `0.3` → `0.6`
- **lambda_http**: `0.8.4` → `1` (for AWS Lambda support)

Updated `aws-smithy-types` to use `http-body-1-x` feature instead of `http-body-0-4-x`.

### 2. New `serve` Module
Added a comprehensive `serve` module (inspired by `axum::serve`) that provides:
- **Simple server API**: Ergonomic `serve(listener, service)` function
- **Connection limiting**: Built-in `limit_connections()` via`ListenerExt`
- **Graceful shutdown**: Zero-cost when unused, opt-in with `.with_graceful_shutdown()`
- **Hyper customization**: `.configure_hyper()` for protocol and performance tuning

## Migration Impact

**Breaking Change**: This updates the public API to use `http@1.x` types (`http::Request`, `http::Response`, etc.).

Commit:
Update examples to use http-1
